### PR TITLE
Fix docker test dependencies

### LINKED ISSUE #2344: Docker tests are failing
[Here](https://github.com/open-telemetry/opentelemetry-python-contrib/actions/runs/8258035377/job/22589628504?pr=2284)

### PR DESCRIPTION
**Pull request by ocelotl**
Fixes #2344

### CODE REVIEW

**Review by xrmx — COMMENTED** [review]
Oops
ERROR: Cannot install aiopg==1.2.1, async-timeout==3.0.1 and redis==4.5.3 because these package versions have conflicting dependencies.

File: tox.ini [hunk 1]
```diff
@@ -597,30 +597,76 @@ commands =
 [testenv:docker-tests]
 basepython: python3
 deps =
-  pip >= 20.3.3
-  pytest
+  aiopg==1.2.1
```
Comment (xrmx) [inline tox.ini:600]:
```suggestion
  aiopg==1.4.0
```

Comment (xrmx) [reply tox.ini:600]: It should be fine, instrumentation supports aiopg < 2

Comment (ocelotl) [reply tox.ini]: Fixed

**Review by pmcollins — APPROVED** [review]

**Pull request merged**